compiler/rustc_lint/src/types.rs

```diff
@@ -1464,19 +1464,16 @@ impl<'a, 'tcx> ImproperCTypesVisitor<'a, 'tcx> {
                 }]);
             }
             ffir @ FfiResult::FfiUnsafeWrapper { .. } => {
-                let mut last_ty = None;
-                let mut ffiresult_recursor = Some(&ffir);
+                let mut ffiresult_recursor = ControlFlow::Continue(&ffir);
                 let mut cimproper_layers: Vec<ImproperCTypesLayer<'tcx>> = vec![];
 
                 // this whole while block converts the arbitrarily-deep
-                // FfiResult stack to a ImproperCTypesLayer Vec
-                while let Some(ref ffir_rec) = ffiresult_recursor {
+                // FfiResult stack to an ImproperCTypesLayer Vec
+                while let ControlFlow::Continue(ref ffir_rec) = ffiresult_recursor {
                     match ffir_rec {
                         FfiResult::FfiPhantom(ty) => {
-                            last_ty = Some(ty.clone());
-                            let len = cimproper_layers.len();
-                            if len > 0 {
-                                cimproper_layers[len - 1].inner_ty = last_ty.clone();
+                            if let Some(layer) = cimproper_layers.last_mut() {
+                                layer.inner_ty = Some(ty.clone());
```

Comment:
I pushed my big style nit as the latest commit.

Reply:
it's definitely okay by me!
...I can't believe I forgot `.last()` and co existed.